src/utils/strcpy_utils.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ *  Copyright (c) 2017-2019, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef OTBR_STRCPY_UTILS_HPP
+#define OTBR_STRCPY_UTILS_HPP
+
+#include <string.h>
+
+/**
+ * This method is a safe version of strncpy always ensuring last byte to be 0
+ * If strlen(aSrc) >= n, the string will be truncated
+ *
+ * @param[in]  aDest      dest string
+ * @param[in]  aSrc       src string
+ * @param[in]  aDestSize  size of dest string buffer
+ *
+ * @returns 0 on success, -1 on error
+ *
+ */
+extern "C" int strncpy_safe(char *aDest, const char *aSrc, size_t aDestSize);
```

Comment:
How about implementing a standard function, either `strcpy_s` or `strncpy_s`, so that once we switched to `C11`, we can simply rename the function.

Reply:
Already implemented to be compatible with `strcpy_s`, use this name just to avoid potential linking issues.